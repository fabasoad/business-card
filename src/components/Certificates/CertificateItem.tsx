import * as React from 'react'
import { Image, Toast } from 'react-bootstrap'
import { useTranslation } from 'react-i18next'
import DateLocale from '../controls/DateLocale'
import { Technology } from '../../scripts/technologies/types'
import { CertificateIssuer } from '../../scripts/certificates/types'

interface CertificateItemProps {
  id: string
  issueDate: Date
  issuer: CertificateIssuer
  technology: Technology
  i18nTitleKey: string
  url: string
}

export default function CertificateItem(props: CertificateItemProps) {
  const { t } = useTranslation()
  return (
    <Toast className="mb-4 mx-2 w-100">
      <Toast.Header closeButton={false}>
        {props.issuer.img &&
          <Image src={props.issuer.img} className="mr-2" rounded />}
        <strong className="mr-2">{props.issuer.name}</strong>
        <DateLocale className="mr-auto" month={props.issueDate.getMonth() + 1} year={props.issueDate.getFullYear()} />
        <small>{props.id}</small>
        <Image className="ml-2" src={props.technology.img} />
      </Toast.Header>
      <Toast.Body className="text-center">
        <a target="_blank" rel="noopener noreferrer" href={props.url}>
          {t(props.i18nTitleKey)}
        </a>
      </Toast.Body>
    </Toast>
  )
}
