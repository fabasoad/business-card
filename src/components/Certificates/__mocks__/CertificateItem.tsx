import * as React from 'react'
import { CertificateItemProps } from '../CertificateItem'

export default function CertificateItem({ certificate }: CertificateItemProps) {
  return (
    <div data-testid={`CertificateItem-${certificate.id}`}></div>
  )
}
