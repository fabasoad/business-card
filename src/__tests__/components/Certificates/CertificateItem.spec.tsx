/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import CertificateItem from '../../../components/Certificates/CertificateItem'

test('should render CertificateItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(<CertificateItem
    id="test-id"
    issueDate={new Date('2020-01')}
    issuer={{ img: 'test-issuer-img', name: 'test-issuer-name' }}
    technology={{ name: 'test-technology-name', title: 'test-technology-title', img: 'test-technology-img' }}
    i18nTitleKey='test-title-key'
    url='test-url'
  />)
  expect(wrapper).toMatchSnapshot()
})
