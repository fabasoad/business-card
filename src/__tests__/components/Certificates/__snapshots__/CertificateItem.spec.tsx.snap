// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CertificateItem correctly 1`] = `
<Toast
  className="mb-4 mx-2 w-100"
>
  <ToastHeader
    closeButton={false}
    closeLabel="Close"
  >
    <Image
      className="mr-2"
      fluid={false}
      rounded={true}
      roundedCircle={false}
      src="test-issuer-img"
      thumbnail={false}
    />
    <strong
      className="mr-2"
    >
      test-issuer-name
    </strong>
    <Connect(DateLocale)
      className="mr-auto"
      month={1}
      year={2020}
    />
    <small>
      test-id
    </small>
    <Image
      className="ml-2"
      fluid={false}
      rounded={false}
      roundedCircle={false}
      src="test-technology-img"
      thumbnail={false}
    />
  </ToastHeader>
  <ToastBody
    className="text-center"
  >
    <a
      href="test-url"
      rel="noopener noreferrer"
      target="_blank"
    >
      test-title-key
    </a>
  </ToastBody>
</Toast>
`;
